heareval/tasks/util/audio.py

```diff
@@ -69,3 +76,39 @@ def resample_wav(in_file: str, out_file: str, out_sr: int):
     )
     # Make sure the return code is 0 and the command was successful.
     assert ret == 0
+
+
+def audio_stats_wav(in_file: Union[str, Path]):
+    """Get statistics for a single wav file"""
+    audio = sf.SoundFile(in_file)
+    return {
+        "samples": len(audio),
+        "sample_rate": audio.samplerate,
+        "duration": round(len(audio) / audio.samplerate, 2),
+    }
+
+
+def audio_dir_stats_wav(in_dir: Union[str, Path], out_file: str):
+    """Produce summary by recursively searching a directory for wav files"""
+
+    audio_paths = list(Path(in_dir).absolute().rglob("*.wav"))
+    audio_dir_stats = list(map(audio_stats_wav, audio_paths))
```

Comment:
Is there a chance this will take a while to run on larger sets? If so a tqdm progress bar would be great.